### PR TITLE
add networking routes to community screen

### PR DESCRIPTION
**Pull request by amycshuang**
## Overview

- Added networking to the community screen for searching users and displaying all users

## Changes Made

- Added networking for searching by name or netid 
- Added networking to display all users

## Test Coverage

- Manual testing with simulator

## Screenshots

<details>

  <summary>Screen Shots</summary>

  

Simulator Screen Shot - iPhone 11 Pro - 2020-11-11 at 19 50 04

Simulator Screen Shot - iPhone 11 Pro - 2020-11-11 at 19 50 23
  

</details>

### CODE REVIEW

File: CoffeeChat/Controller/CommunityViewController.swift [hunk 1]
```diff
@@ -15,63 +15,24 @@ class CommunityViewController: UIViewController {
...
-            matches: nil,
-            netID: "llx2",
-            profilePictureURL: "",
-            pronouns: "He/Him",
-            facebook: nil,
-            instagram: nil
-        )
-    ]
-
+    private var users: [CommunityUser] = []
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/CommunityViewController.swift:18]: Yay! Love seeing this get deleted

File: CoffeeChat/Controller/CommunityViewController.swift [hunk 2]
```diff
@@ -15,63 +15,24 @@ class CommunityViewController: UIViewController {
...
     override func viewDidLoad() {
         super.viewDidLoad()
 
         view.backgroundColor = .backgroundLightGreen
 
+        NetworkManager.shared.getUsers().observe { (response) in
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/CommunityViewController.swift:24]: don't need parentheses around response

File: CoffeeChat/Controller/CommunityViewController.swift [hunk 3]
```diff
@@ -15,63 +15,24 @@ class CommunityViewController: UIViewController {
...
+            switch response {
+            case .value(let value):
+                print("success")
+                print(value.success)
+                print(value.data)
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/CommunityViewController.swift:29]: Are we doing anything with this data?

File: CoffeeChat/Controller/CommunityViewController.swift [hunk 4]
```diff
@@ -15,63 +15,24 @@ class CommunityViewController: UIViewController {
...
+            case .error(let error):
+                print("error")
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/CommunityViewController.swift:31]: Can probably delete this

File: CoffeeChat/Controller/CommunityViewController.swift (hunk 2 above)
Comment (lucyxubroad) [reply CoffeeChat/Controller/CommunityViewController.swift:24]: Oh I guess this is not needed, so delete?

File: CoffeeChat/Controller/CommunityViewController.swift [hunk 5]
```diff
@@ -99,8 +60,24 @@ class CommunityViewController: UIViewController {
         communityTableView.register(CommunityUserTableViewCell.self, forCellReuseIdentifier: CommunityUserTableViewCell.reuseIdentifier)
         view.addSubview(communityTableView)
 
+        getUsers()
         setupConstraints()
     }
+
+    private func getUsers() {
+        NetworkManager.shared.getUsers().observe { (response) in
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/CommunityViewController.swift:68]: remove parentheses around `response`

File: CoffeeChat/Controller/CommunityViewController.swift [hunk 6]
```diff
@@ -137,8 +114,28 @@ extension CommunityViewController: UITableViewDataSource {
 // MARK: - SearchBarDelegate
 extension CommunityViewController: UISearchBarDelegate {
 
+    private func searchUsers(query: String) {
+        NetworkManager.shared.searchUsers(query: query).observe { (response) in
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/CommunityViewController.swift:118]: remove parentheses around response

File: CoffeeChat/Controller/CommunityViewController.swift [hunk 7]
```diff
@@ -137,8 +114,28 @@ extension CommunityViewController: UITableViewDataSource {
...
+            switch response {
+            case .value(let value):
+                guard value.success else { return }
+                DispatchQueue.main.async {
+                    self.users = value.data
+                    self.communityTableView.reloadData()
+                }
+            case .error(let error):
+                print("error")
+                print(error)
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/CommunityViewController.swift:128]: remove print("error")?

File: CoffeeChat/Controller/CommunityViewController.swift [hunk 8]
```diff
@@ -137,8 +114,28 @@ extension CommunityViewController: UITableViewDataSource {
...
+            }
+        }
+    }
+
     func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
-        // TODO: Implement search and filter
+        if searchText.count < 1 {
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/CommunityViewController.swift:134]: can also do == 0?

**Review by lucyxubroad — APPROVED** [review]
This looks fantastic! Great of you to get this done! :)

File: CoffeeChat/Controller/CommunityViewController.swift (hunk 8 above)
Comment (pnor) [reply CoffeeChat/Controller/CommunityViewController.swift:134]: or `searchText.isEmpty`

**Review by pnor — APPROVED** [review]
Looks great!

**Pull request merged**